include/deal.II/base/types.h

```diff
@@ -173,6 +173,14 @@ namespace numbers
   static const unsigned int invalid_unsigned_int =
     static_cast<unsigned int>(-1);
 
+  /**
+   * Representation of the largest number that can be put into an unsigned
+   * long integer.
+   */
+  static const unsigned long int invalid_unsigned_long_int =
+    static_cast<unsigned long int>(-1);
```

Comment:
Then you can name this here `invalid_coarse_cell_id` and define it in the same place as `invalid_subdomain_id` and the other `invalid_...` numbers.